### PR TITLE
Proxy/implementation

### PR DESCRIPTION
**Pull request by Tixwell**
I just finish the deploy proxy and a test to upgrade it, I will wrote the deploy-proxy / upgrade-proxy script once you finish change into hardhat 

cc @azerpas

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mnemonic=....
+alchemy_goerli_url=....
```
Comment (azerpas) [inline .env.example:2]: In UPPERCASE bro

File: README.md [hunk 2]
```diff
@@ -9,6 +9,11 @@
 - [ganache](https://www.npmjs.com/package/ganache)
 - [truffle](https://www.npmjs.com/package/truffle)
 ### Setup
+- Setup env keys
+```sh
+mnemonic=....
+alchemy_goerli_url=....
+```
 - Run in shell
```
Comment (azerpas) [inline README.md:17]: nice

File: contracts/FractifV1.sol [hunk 3]
```diff
@@ -92,7 +92,7 @@ contract FractifV1 is
                 "The amount of ETH sent should be greater than or equal to the sellout price"
             );
         } else {
-            IERC20(depositedCoin).transferFrom(
+            IERC20Upgradeable(depositedCoin).transferFrom(
```
Comment (azerpas) [inline contracts/FractifV1.sol:95]: This shouldn't get converted to `IERC20Upgradeable` as we're only using IERC to transfer tokens from the smart-contract balance to users, but these tokens on their own we don't know if they're upgradable. 

**same goes for the same changes below 1️⃣**

File: contracts/FractifV1.sol [hunk 4]
```diff
@@ -158,7 +158,7 @@ contract FractifV1 is
         }
         // ERC20
         else {
-            uint256 balance = IERC20(tokenDepositedCoin[tokenId]).balanceOf(
+            uint256 balance = IERC20Upgradeable(tokenDepositedCoin[tokenId]).balanceOf(
```
Comment (azerpas) [inline contracts/FractifV1.sol:161]: 1️⃣

File: contracts/FractifV1.sol [hunk 5]
```diff
@@ -167,7 +167,7 @@ contract FractifV1 is
                     refundAmount: refundAmount
                 });
             } else {
-                IERC20(tokenDepositedCoin[tokenId]).transfer(
+                IERC20Upgradeable(tokenDepositedCoin[tokenId]).transfer(
```
Comment (azerpas) [inline contracts/FractifV1.sol:170]: 1️⃣

**Review by azerpas — CHANGES_REQUESTED** [review]
Nice, can you review the changes with IERC20Upgradable? Otherwise it should be good!

**Comment by Tixwell** [comment]
@azerpas tell me if those change are good for you and tell me when I can merge it!

**Review by azerpas — APPROVED** [review]
LGTM :)

**Pull request merged**